MobiusCore/Source/ConnectableConvenienceClasses/AsyncDispatchQueueConnectable.swift

```diff
@@ -44,17 +54,33 @@ final class AsyncDispatchQueueConnectable<InputType, OutputType>: Connectable {
     }
 
     func connect(_ consumer: @escaping (OutputType) -> Void) -> Connection<InputType> {
-        let connection = underlyingConnectable.connect(consumer)
+        let connection = underlyingConnectable.connect { [disposalStatus] value in
+            // Don’t forward if we’re currently waiting to dispose the connection.
+            //
+            // NOTE: the underlying consumer must be called inside the criticial region accessing disposalStatus, or we
+            // could potentially enter the .pendingDispose state before or during the consumer call. This means that the
+            // underlying consumer must not call our connection’s acceptClosure or disposeClosure, or it will deadlock.
+            //
+            // This is OK in our existing use case because the underlying consumer is always flipEventsToLoopQueue from
+            // MobiusController’s initializer, which enters the actual Mobius loop asynchronously. If we exposed this
+            // class, it would be a scary edge case.
+            disposalStatus.read { status in
+                guard status != .pendingDispose else { return }
+                consumer(value)
+            }
+        }
 
         return Connection(
             acceptClosure: { [acceptQueue] input in
                 acceptQueue.async {
                     connection.accept(input)
                 }
             },
-            disposeClosure: { [acceptQueue] in
+            disposeClosure: { [acceptQueue, disposalStatus] in
+                disposalStatus.value = .pendingDispose
```

Comment:
What happens if there are multiple calls to `dispose`?

Reply:
Updated to assert on that… and scope the `disposalStatus` to the individual connection, not the connectable. Oops!